features/FEATURE_PSA/TARGET_TFM/src/os_wrapper_cmsis_rtos_v2.cpp

```diff
@@ -0,0 +1,168 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "os_wrapper/thread.h"
+#include "os_wrapper/mutex.h"
+#include "os_wrapper/semaphore.h"
+#include "rtos/Thread.h"
+#include "rtos/ThisThread.h"
+#include "rtos/Semaphore.h"
+#include "rtos/Mutex.h"
+
+extern "C" void *os_wrapper_semaphore_create(uint32_t max_count, uint32_t initial_count,
+                                             const char *name)
+{
+    if (initial_count > max_count) {
+        return NULL;
+    }
+
+    return (void *)new rtos::Semaphore(initial_count, max_count);
+}
+
+extern "C" uint32_t os_wrapper_semaphore_acquire(void *handle, uint32_t timeout)
+{
+    bool status;
+    rtos::Semaphore *new_sema = (rtos::Semaphore *)handle;
+
+    status = new_sema->try_acquire_for((timeout == OS_WRAPPER_WAIT_FOREVER) ?
+                                       osWaitForever : timeout);
+    if (false == status) {
+        return OS_WRAPPER_ERROR;
+    }
+
+    return OS_WRAPPER_SUCCESS;
+}
+
+extern "C" uint32_t os_wrapper_semaphore_release(void *handle)
+{
+    osStatus status;
+    rtos::Semaphore *new_sema = (rtos::Semaphore *)handle;
+
+    status = new_sema->release();
+    if (status != osOK) {
+        return OS_WRAPPER_ERROR;
+    }
+
+    return OS_WRAPPER_SUCCESS;
+}
+
+extern "C" uint32_t os_wrapper_semaphore_delete(void *handle)
+{
+    delete (rtos::Semaphore *)handle;
+
+    return OS_WRAPPER_SUCCESS;
+}
+
+extern "C" void *os_wrapper_mutex_create(void)
+{
+    return (void *)new rtos::Mutex();
+}
+
+extern "C" uint32_t os_wrapper_mutex_acquire(void *handle, uint32_t timeout)
+{
+    bool status;
+    rtos::Mutex *new_mutex = (rtos::Mutex *)handle;
+
+    status = new_mutex->trylock_for((timeout == OS_WRAPPER_WAIT_FOREVER) ?
+                                    osWaitForever : timeout);
+    if (!status) {
+        return OS_WRAPPER_ERROR;
+    }
+
+    return OS_WRAPPER_SUCCESS;
+}
+
+extern "C" uint32_t os_wrapper_mutex_release(void *handle)
+{
+    osStatus status = osOK;
+    rtos::Mutex *new_mutex = (rtos::Mutex *)handle;
+
+    status = new_mutex->unlock();
+    if (status != osOK) {
+        return OS_WRAPPER_ERROR;
+    }
+
+    return OS_WRAPPER_SUCCESS;
+}
+
+extern "C" uint32_t os_wrapper_mutex_delete(void *handle)
+{
+    delete (rtos::Mutex *)handle;
```

Comment:
Make sure that semaphore is released before freeing allocated memory.

Reply:
I think the same comment applies here - https://github.com/ARMmbed/mbed-os/pull/12271#discussion_r375267774

Reply:
According to this - https://github.com/ARMmbed/mbed-os/blob/feature_twincpu/rtos/source/TARGET_CORTEX/rtx5/RTX/Source/rtx_mutex.c#L376
should not be problem, if delete is called.